rescompy/rescompy.py

```diff
@@ -209,65 +208,126 @@ class TrainResult:
+							  (states[task_ind].shape[0], None), 'inputs')
+            utils.check_shape(weights.shape,
+							  (features.shape[1], target_outputs[0].shape[1]),
+							  'weights')
+            if (target_outputs[task_ind].shape[0] > 1):
+                utils.check_shape(target_outputs[task_ind].shape,
+								  (states[task_ind].shape[0], None), 'target_outputs')
+
+            else:
+                utils.check_range(features.shape[0],
+								  "The number of feature vectors per input signal",
+								  1, 'eq', True)
         
         # Assign attributes.
-        self.states = states
-        self.inputs = inputs
-        self.target_outputs = target_outputs
         self.feature_function = feature_function
+        self.accessible_drives = accessible_drives
         self.weights = weights
-        self.transient_length = transient_length
         self.jacobian = jacobian
         
+		# Check if the feature function requires a lookback period to include
+        # time-delayed states or inputs in feature vectors.
+        if hasattr(feature_function, 'lookback_length'):
+            self.lookback_length = feature_function.lookback_length
+        else: self.lookback_length = 0
+
+        self.listed_targets = target_outputs		
+        self.listed_transients = transient_lengths
+        self.listed_inputs = inputs
+        self.listed_states = states
+		
+        self.states = None
+        for task_ind in range(len(self.listed_inputs)):
+            if (self.states is None):
+                self.states = self.listed_states[task_ind]
+                self.inputs = self.listed_inputs[task_ind]
+                self.target_outputs = self.listed_targets[task_ind]
+            else:
+                self.states = np.concatenate((self.states,
+											  self.listed_states[task_ind]))
+                self.inputs = np.concatenate((self.inputs,
+											  self.listed_inputs[task_ind]))
+                self.target_outputs = np.concatenate((self.target_outputs,
+													  self.listed_targets[task_ind]))
+						
+    @property
+    def transient_length(self):
+        """The transient length associated with the TrainResult object.
+        If the same transient length applied to all training signals, this 
+        this transient length is returned.
+        If multiple transient lengths applied, an error will be raised."""
+        
+        transients = np.array(self.listed_transients)
+        if transients == transients[0]:
+            return transients[0]
+        else:
+            msg = "Transient length is not defined since different training " \
+                      "tasks employed different transient lengths."
+            logging.error(msg)
+        
     @property
     def features(self):
         """The features property.
         Computes the reservoir features from the states and inputs using the
         feature_function."""
-        return self.feature_function(self.states, self.inputs)
+        features = None
```

Comment:
maybe this should be done with a `functools.reduce()`?

Reply:
Done